grype/pkg/package.go

```diff
@@ -135,14 +135,29 @@ func excludePackage(p pkg.Package, parent pkg.Package) bool {
 		return false
 	}
 
+	// If the parent is an OS package and the child is not, exclude the child
+	if isOSPackage(parent) && !isOSPackage(p) {
+		return true
+	}
+
 	// filter out only binary pkg, empty types, or equal types
 	if p.Type != pkg.BinaryPkg && p.Type != "" && p.Type != parent.Type {
+		// when I fix the version info, I think this one will exclude it.
 		return false
 	}
 
 	return true
 }
 
+func isOSPackage(p pkg.Package) bool {
+	switch p.Type {
+	case pkg.DebPkg, pkg.RpmPkg, pkg.ApkPkg:
```

Comment:
It's not clear to me exactly which package types this should be true on. These seem like a reasonable start.